platform/kube/inject/inject.go

```diff
@@ -137,46 +140,71 @@ func injectIntoPodTemplateSpec(p *Params, t *v1.PodTemplateSpec) error {
 	t.Annotations["pod.beta.kubernetes.io/init-containers"] = string(initAnnotationValue)
 
 	// sidecar proxy container
-	t.Spec.Containers = append(t.Spec.Containers,
-		v1.Container{
-			Name:  proxyContainerName,
-			Image: p.ProxyImage,
-			Args: []string{
-				"proxy",
-				"sidecar",
-				"-s", p.ManagerAddr,
-				"-m", p.MixerAddr,
-				"-n", "$(POD_NAMESPACE)",
-				"-v", strconv.Itoa(p.Verbosity),
+	args := []string{
+		"proxy",
+		"sidecar",
+		"-s", p.ManagerAddr,
+		"-m", p.MixerAddr,
+		"-n", "$(POD_NAMESPACE)",
+		"-v", strconv.Itoa(p.Verbosity),
+	}
+	var volumeMounts []v1.VolumeMount
+	if p.EnableAuth {
+		args = append(args, "--enable_auth", "--auth_config_path", p.AuthConfigPath)
+		volumeMounts = append(volumeMounts, v1.VolumeMount{
+			Name:      istioCertVolumeName,
+			ReadOnly:  true,
+			MountPath: p.AuthConfigPath,
+		})
+
+		sa := t.Spec.ServiceAccountName
+		if sa == "" {
+			sa = "default"
```

Comment:
I'm somewhat concerned that you are making an assumption about the name of the service account. This script can be run before AC injects the service account. What happens with deployments? Do pods share the same service account?